### ISSUE TITLE
RTVS locks package dlls, interferes with package (re)installation

### ISSUE THREAD

**Issue by Hong-Revo** [Microsoft/RTVS#3467]
I think there was an existing bug report on this, but I can't find it. RTVS seems to lock a bunch of dlls, which stops R from overwriting them as part of package reinstallation. I just ran into this when trying to install RSQLServer from Github. The process broke on installing the rlang dependency:

```
Installing rlang
"C:/PROGRA~1/MICROS~3/RSERVE~1/R_SERVER//bin/x64/R" --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  "C:/Users/hong/AppData/Local/Temp/RtmpgDMvve/devtools118c13d86b3b/hadley-rlang-0756eaf" --library="c:/Rlib" --install-tests 

* installing *source* package 'rlang' ...
** libs
c:/Rtools/mingw_64/bin/gcc -m64 -I"C:/PROGRA~1/MICROS~3/RSERVE~1/R_SERVER/include" -DNDEBUG     -I"C:/swarm/workspace/External-R-3.3.3/vendor/extsoft/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c attrs.c -o attrs.o
c:/Rtools/mingw_64/bin/gcc -m64 -I"C:/PROGRA~1/MICROS~3/RSERVE~1/R_SERVER/include" -DNDEBUG     -I"C:/swarm/workspace/External-R-3.3.3/vendor/extsoft/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c capture.c -o capture.o
c:/Rtools/mingw_64/bin/gcc -m64 -I"C:/PROGRA~1/MICROS~3/RSERVE~1/R_SERVER/include" -DNDEBUG     -I"C:/swarm/workspace/External-R-3.3.3/vendor/extsoft/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c eval.c -o eval.o
c:/Rtools/mingw_64/bin/gcc -m64 -I"C:/PROGRA~1/MICROS~3/RSERVE~1/R_SERVER/include" -DNDEBUG     -I"C:/swarm/workspace/External-R-3.3.3/vendor/extsoft/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c internals.c -o internals.o
c:/Rtools/mingw_64/bin/gcc -m64 -I"C:/PROGRA~1/MICROS~3/RSERVE~1/R_SERVER/include" -DNDEBUG     -I"C:/swarm/workspace/External-R-3.3.3/vendor/extsoft/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c pairlist.c -o pairlist.o
c:/Rtools/mingw_64/bin/gcc -m64 -I"C:/PROGRA~1/MICROS~3/RSERVE~1/R_SERVER/include" -DNDEBUG     -I"C:/swarm/workspace/External-R-3.3.3/vendor/extsoft/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c replace-na.c -o replace-na.o
c:/Rtools/mingw_64/bin/gcc -m64 -I"C:/PROGRA~1/MICROS~3/RSERVE~1/R_SERVER/include" -DNDEBUG     -I"C:/swarm/workspace/External-R-3.3.3/vendor/extsoft/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c splice.c -o splice.o
c:/Rtools/mingw_64/bin/gcc -m64 -I"C:/PROGRA~1/MICROS~3/RSERVE~1/R_SERVER/include" -DNDEBUG     -I"C:/swarm/workspace/External-R-3.3.3/vendor/extsoft/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c symbol.c -o symbol.o
c:/Rtools/mingw_64/bin/gcc -m64 -I"C:/PROGRA~1/MICROS~3/RSERVE~1/R_SERVER/include" -DNDEBUG     -I"C:/swarm/workspace/External-R-3.3.3/vendor/extsoft/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c unquote.c -o unquote.o
c:/Rtools/mingw_64/bin/gcc -m64 -I"C:/PROGRA~1/MICROS~3/RSERVE~1/R_SERVER/include" -DNDEBUG     -I"C:/swarm/workspace/External-R-3.3.3/vendor/extsoft/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c utils.c -o utils.o
c:/Rtools/mingw_64/bin/gcc -m64 -shared -s -static-libgcc -o rlang.dll tmp.def attrs.o capture.o eval.o internals.o pairlist.o replace-na.o splice.o symbol.o unquote.o utils.o -LC:/swarm/workspace/External-R-3.3.3/vendor/extsoft/lib/x64 -LC:/swarm/workspace/External-R-3.3.3/vendor/extsoft/lib -LC:/PROGRA~1/MICROS~3/RSERVE~1/R_SERVER/bin/x64 -lR
installing to c:/Rlib/rlang/libs/x64
Warning in file.copy(files, dest, overwrite = TRUE) :
  problem copying .\rlang.dll to c:\Rlib\rlang\libs\x64\rlang.dll: Permission denied
** R
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
Error in FUN(X[[i]], ...) : 
  no such symbol rlang_splice in package c:/Rlib/rlang/libs/x64/rlang.dll
Error: loading failed
Execution halted
ERROR: loading failed
```

It works when I run it from rgui.

**Comment by Hong-Revo** [comment]
This is with RTVS 1.1.30423.1752, on local and remote.

**Comment by MikhailArkhipov** [comment]
there was a bug on this, the fix improved the case quite a bit, but occasionally it does still happen.

**Comment by MikhailArkhipov** [comment]
Original issue was that installation of packages triggered rebuild of intellisense index and the rebuild loads packages and locks them causing conflicts. A notification was added so index would only start rebuilding after all dependent packages installed. However, this only applies to a single install.packages command (and its dependencies). If one executes multiple in sequence index update theoretically can start between commands. We can add some more throttling, but it won't be complete solution.

**Comment by Hong-Revo** [comment]
Would it be feasible to check the status of the REPL, and only rebuild the index if the command prompt is active? I dunno, just throwing ideas out there.

**Comment by MikhailArkhipov** [comment]
Probably. May be possible to fix along with #3474

**Comment by MikhailArkhipov** [comment]
https://github.com/Microsoft/RTVS/commit/b4dc0762f6f834a72930255df499d7197ba264f3
